res/css/views/rooms/_SearchBar.pcss

```diff
@@ -15,30 +15,43 @@ limitations under the License.
 */
 
 .mx_SearchBar {
-    height: 56px;
+    min-height: 56px; /* for wrapped mx_SearchBar_button making the area tall */
     display: flex;
     align-items: center;
     border-bottom: 1px solid $primary-hairline-color;
 
     .mx_SearchBar_input {
+        --size-button-search: 37px;
+
         /* border: 1px solid $input-border-color; */
         /* font-size: $font-15px; */
         flex: 1 1 0;
         margin-left: 22px;
+
+        /* retain space for the input bar. 7em is a magic number to retain space for the placeholder 'Search…' */
+        min-width: calc(7em + var(--size-button-search));
+
+        input {
+            box-sizing: border-box; /* include padding value into width calculation */
+        }
     }
 
     .mx_SearchBar_searchButton {
+        --size: var(--size-button-search); /* square */
+
         cursor: pointer;
-        width: 37px;
-        height: 37px;
+        width: var(--size);
+        min-width: var(--size);
```

Comment:
why do we need `min-width` as well as `width`?

Reply:
I might be confused with something else. Now min-width for `input` is secured, we do not need this.